src/biogeophys/InfiltrationExcessRunoffMod.F90

```diff
@@ -0,0 +1,342 @@
+    real(r8) , intent(in)    :: qflx_in_soil( bounds%begc: )               ! surface input to soil (mm/s)
+    real(r8) , intent(inout) :: qinmax_on_unsaturated_area( bounds%begc: ) ! maximum infiltration capacity on the unsaturated fraction of the column (mm H2O /s)
+    !
+    ! !LOCAL VARIABLES:
+    integer  :: fc, c
+    real(r8) :: dtime       ! land model time step (sec)
+    real(r8) :: top_icefrac ! ice fraction in top VIC layers
+    real(r8) :: max_infil   ! max infiltration capacity in VIC (mm)
+    real(r8) :: i_0         ! average soil moisture in top VIC layers (mm)
+    real(r8) :: rsurf_vic   ! VIC surface runoff
+    real(r8) :: basis       ! variable soil moisture holding capacity in top VIC layers for runoff calculation
+
+    character(len=*), parameter :: subname = 'ComputeQinmaxVic'
+    !-----------------------------------------------------------------------
+
+    SHR_ASSERT_ALL((ubound(fsat) == (/bounds%endc/)), errMsg(sourcefile, __LINE__))
+    SHR_ASSERT_ALL((ubound(qflx_in_soil) == (/bounds%endc/)), errMsg(sourcefile, __LINE__))
+    SHR_ASSERT_ALL((ubound(qinmax_on_unsaturated_area) == (/bounds%endc/)), errMsg(sourcefile, __LINE__))
+
+    associate( &
+          top_max_moist    =>    soilhydrology_inst%top_max_moist_col, & ! Input:  [real(r8) (:)   ]  maximum soil moisture in top VIC layers
+          top_moist        =>    soilhydrology_inst%top_moist_col    , & ! Input:  [real(r8) (:)   ]  soil moisture in top VIC layers
+          top_ice          =>    soilhydrology_inst%top_ice_col      , & ! Input:  [real(r8) (:)   ]  ice len in top VIC layers
+          b_infil          =>    soilhydrology_inst%b_infil_col        & ! Input:  [real(r8) (:)   ]  VIC b infiltration parameter
+          )
+
+    dtime = get_step_size()
+
+    do fc = 1, num_hydrologyc
+       c = filter_hydrologyc(fc)
+       top_icefrac = min(1._r8,top_ice(c)/top_max_moist(c))
+       max_infil = (1._r8+b_infil(c)) * top_max_moist(c)
+       i_0 = max_infil * (1._r8 - (1._r8 - fsat(c))**(1._r8/b_infil(c)))
+       if(qflx_in_soil(c) <= 0._r8) then
+          rsurf_vic = 0._r8
+       else if(max_infil <= 0._r8) then
+          rsurf_vic = qflx_in_soil(c)
+       else if((i_0 + qflx_in_soil(c)*dtime) > max_infil) then             !(Eq.(3a) Wood et al. 1992)
+          rsurf_vic = (qflx_in_soil(c)*dtime - top_max_moist(c) + top_moist(c))/dtime
+       else                                                                !(Eq.(3b) Wood et al. 1992)
+          basis = 1._r8 - (i_0 + qflx_in_soil(c)*dtime)/max_infil
+          rsurf_vic = (qflx_in_soil(c)*dtime - top_max_moist(c) + top_moist(c)    &
+               + top_max_moist(c) * basis**(1._r8 + b_infil(c)))/dtime
+       end if
+       rsurf_vic = min(qflx_in_soil(c), rsurf_vic)
+       qinmax_on_unsaturated_area(c) = 10._r8**(-e_ice*top_icefrac)*(qflx_in_soil(c) - rsurf_vic)
```

Comment:
**Comment from @martynpclark Aug 24, 2017**

This single line of code is actually infiltration excess runoff -- adding ice impedance -- though infiltration excess runoff was not in the original VIC model.

We could have different decisions for infiltration excess and saturation excess runoff, so it is possible to still include these capabilities.